src/libpriv/rpmostree-jigdo-client.c

```diff
@@ -42,18 +42,21 @@
 
 static DnfPackage *
 query_jigdo_pkg (DnfContext *dnfctx,
-                 const char *name,
+                 const char *name_arch,
                  const char *evr,
                  GError    **error)
 {
   hy_autoquery HyQuery query = hy_query_create (dnf_context_get_sack (dnfctx));
-  /* See also similar examples of queries in e.g. dnf_context_update() */
-  hy_query_filter (query, HY_PKG_NAME, HY_EQ, name);
-  hy_query_filter (query, HY_PKG_ARCH, HY_NEQ, "src");
+  /* This changed in v4, we now go through the Provides: name(arch) */
+  const gboolean is_archful = strchr (name_arch, '(') != NULL;
+  if (is_archful)
+    hy_query_filter (query, HY_PKG_PROVIDES, HY_EQ, name_arch);
+  else
+    hy_query_filter (query, HY_PKG_NAME, HY_EQ, name_arch);
```

Comment:
Given that `(` is an illegal character in names, let's extract the name in the archful case and do this unconditionally?

Reply:
I'm not understanding what you mean here.  Like always query `HY_PKG_NAME`?  That wouldn't work as it's a `_PROVIDES`.

Reply:
Basically if it's archful, we can just extract the name from `name_arch` and still add a `HY_PKG_NAME` filter, right? (On top of the existing `PROVIDES`).

Reply:
OK, done :arrow_down: